event-handlers/src/main/java/no/sikt/nva/nvi/events/batch/RequeueDlqHandler.java

```diff
@@ -0,0 +1,170 @@
+    public static final String REQUEUE_DLQ_STARTED_LOG = "Requeue DLQ started. {} messages to process.";
+    public static final String DELETING_MESSAGE_FROM_DLQ_LOG = "Deleting message from DLQ: {}";
+    public static final String PROCESSING_MESSAGE_LOG = "Processing message: {}";
+    public static final String CANDIDATE_IDENTIFIER_ATTRIBUTE_NAME = "candidateIdentifier";
+    public static final String COULD_NOT_UPDATE_CANDIDATE = "Could not update candidate: %s";
+    public static final String COULD_NOT_PROCESS_MESSAGE_LOG = "Could not process message: {}";
+
+    private final NviQueueClient queueClient;
+    private final String queueUrl;
+    private final CandidateRepository candidateRepository;
+    private final PeriodRepository periodRepository;
+
+    @JacocoGenerated
+    public RequeueDlqHandler() {
+        this(
+            new NviQueueClient(),
+            new Environment().readEnv(DLQ_QUEUE_URL_ENV_NAME),
+            new CandidateRepository(defaultDynamoClient()),
+            new PeriodRepository(defaultDynamoClient()));
+    }
+
+    public RequeueDlqHandler(NviQueueClient queueClient, String dlqQueueUrl, CandidateRepository candidateRepository,
+                             PeriodRepository periodRepository) {
+        this.queueClient = queueClient;
+        this.queueUrl = dlqQueueUrl;
+        this.candidateRepository = candidateRepository;
+        this.periodRepository = periodRepository;
+    }
+
+    @Override
+    public RequeueDlqOutput handleRequest(RequeueDlqInput input, Context context) {
+        LOGGER.info(REQUEUE_DLQ_STARTED_LOG, input.count());
+
+        Set<String> messageIdDuplicateCheck = new HashSet<>();
+
+        var remainingMessages = input.count();
+        var failedBatchesCount = 0;
+        var result = new HashSet<NviProcessMessageResult>();
+
+        while (remainingMessages > 0) {
+            var messagesToReceive = Math.min(remainingMessages, MAX_SQS_MESSAGE_COUNT_LIMIT);
+            var response = queueClient.receiveMessage(queueUrl, messagesToReceive);
+
+            if (shouldBreakLoop(response, failedBatchesCount)) {
+                break;
+            }
+
+            var processedMessages = processMessages(response, messageIdDuplicateCheck);
+
+            result.addAll(processedMessages);
+            failedBatchesCount += checkForFailedBatch(processedMessages);
+            remainingMessages -= messagesToReceive;
+        }
+
+        LOGGER.info(
+            HANDLER_FINISH_REPORT_LOG,
+            result.size(),
+            result.stream().filter(NviProcessMessageResult::success).count(),
+            result.stream().filter(a -> !a.success()).count(),
+            failedBatchesCount);
+
+        return new RequeueDlqOutput(result, failedBatchesCount);
+    }
+
+    private boolean shouldBreakLoop(NviReceiveMessageResponse response, int failedBatchesCount) {
+        return response.messages().isEmpty() || failedBatchesCount >= MAX_FAILURES;
```

Comment:
An alternative or additional fail check, could be to use the sqs->ReceiveCount for each message, to make sure a message is not handled too many times

Reply:
will not do this at right now